File: KITTY/KURS_1/JAVA/DAY_28_2N_BroadcastReceiver/app/src/main/java/com/example/velic/day_28_broadcastreceiver/TimeBroadcastReceiver.java
package com.example.velic.day_28_broadcastreceiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;
import android.widget.Toast;

import java.text.Format;
import java.text.SimpleDateFormat;
import java.util.Date;

public class TimeBroadcastReceiver extends BroadcastReceiver {
    public TimeBroadcastReceiver() {
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        StringBuilder msgStr = new StringBuilder("Текущее время: ");
        Format formatter = new SimpleDateFormat("hh:mm:ss a");
        msgStr.append(formatter.format(new Date()));
        Toast.makeText(context, msgStr, Toast.LENGTH_SHORT).show();
        Log.d("DAY_28", "onReceive: "+msgStr);
    }
}

File: KITTY/KURS_1/KOTLIN/DAY_28_1_Broadcast_kt/app/src/main/java/com/example/day_28_1_broadcast_k/TimeBroadcastReceiver.kt
package com.example.day_28_1_broadcast_k

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import android.widget.Toast

import java.text.SimpleDateFormat
import java.util.Date

class TimeBroadcastReceiver : BroadcastReceiver() {
        // Регистрируется программно затем слушает время
    override fun onReceive(context: Context, intent: Intent) {
        val msgStr = StringBuilder("Текущее время: ")
        val formatter = SimpleDateFormat("hh:mm:ss a")
        msgStr.append(formatter.format(Date()))
        Toast.makeText(context, msgStr, Toast.LENGTH_SHORT).show()
        Log.d("DAY_28", "onReceive: $msgStr")
    }
}
